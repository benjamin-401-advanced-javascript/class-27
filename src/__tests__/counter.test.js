import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter';

describe('<Thing/>', () => {
  it('is alive at application start', () => {
    const app = shallow(<Counter />);
    expect(app.find('button').exists()).toBeTruthy();
    expect(app.find('span').exists()).toBeTruthy();
  });

  it('changes state on up click', () => {
    const app = mount(<Counter />);
    const button = app.find('.up-clicker');
    button.simulate('click');
    expect(app.state('count')).toBe(1);
    expect(app.find('span').text()).toContain('1');
  });

  it('changes state on down click', () => {
    const app = mount(<Counter />);
    const button = app.find('.down-clicker');
    button.simulate('click');
    expect(app.state('count')).toBe(-1);
    expect(app.find('span').text()).toContain('-1');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
