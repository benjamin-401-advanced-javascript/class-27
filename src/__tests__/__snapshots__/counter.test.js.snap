// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing/> renders correctly 1`] = `
<section
  className="counter"
>
  <button
    className="down-clicker"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="count "
  >
    0
  </span>
  <button
    className="up-clicker"
    onClick={[Function]}
  >
    +
  </button>
</section>
`;
